enhancements/kubelet/oci-image-volume.md

```diff
@@ -0,0 +1,180 @@
+  - N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+# OCI Image Volume
+
+## Summary
+
+In Kubernetes 1.33, support for OCI Image Volume went to Beta.
+This enhancement is to enable this feature in OpenShift.
+See [KEP-4639: Summary](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4639-oci-volume-source#summary)
+
+## Motivation
+
+See [KEP-4639: Motivation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4639-oci-volume-source#motivation)
+
+### User Stories
+
+See [KEP-4639: User Stories](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4639-oci-volume-source#user-stories-optional)
+
+### Goals
+
+- Enable ImageVolume feature gate for OpenShift.
+
+### Non-Goals
+
+- This proposal does not aim to replace existing VolumeSource types.
+- This proposal does not address other use cases for OCI objects beyond directory sharing among containers in a pod.
+- Mounting thousands of images in a single pod.
+
+## Proposal
+
+1. Enable ImageVolume feature gate in 4.20
+
+    - This feature gate is disabled by default in Kubernetes 1.33, so we need to explicitly enable it for OCP
+
+2. Separate Graduation Approach between OCI Images and OCI Artifacts.
+
+    - **OCI Image Mount**: Promote to GA
+    - **OCI Artifact Mount**: Begin as DevPreview
+        - OCI Artifact functionality is less mature and requires further refinement
+        - Add a new CRI-O option to toggle OCI artifact mount support
+        - Disable OCI artifact mount by default
+
+### Workflow Description
+
+#### OCI Image
+
+Since OCI image mount will be GA in 4.20. No feature gate configuration is needed by users. Creating a pod with volumes.image will pull the image and mount in the container.
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod
+spec:
+  containers:
+  - name: test
+    image: registry.k8s.io/e2e-test-images/echoserver:2.3
+    volumeMounts:
+    - name: volume
+      mountPath: /volume
+  volumes:
+  - name: volume
+    image:
+      reference: quay.io/crio/artifact:v1
+      pullPolicy: IfNotPresent
+```
+
+#### OCI Artifact
+
+While OCI artifact is in dev preview, it fails to mount if a user specifies OCI artifact in reference. To enable the mount, users need to add this drop-in CRI-O configuration.
+
+```
+[crio.image]
+oci_artifact_mount_support = true
+```
+
+When it's enabled, users can mount OCI artifacts in the same way as OCI images. Once we feel comfortable with the CRI-O's implementation and make it GA, we'd remove this option.
+
+### API Extensions
+
+No API change is needed on the OCP side.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+To separate graduation of OCI image and OCI artifact, we are going to add a new option to disable OCI artifact in CRI-O and disable OCI artifact mount by default via MCO.
```

Comment:
Would this not be done via an API change?

Reply:
why expose an API we don't expect a user to toggle? Either we're comfortable users using it OOTB or it's off IMO

Reply:
So all the APIs to enable/disable for MCO/Feature gates are present?

Reply:
The feature gate will be enabled by default for OCP.
But because we disable OCI artifacts in MCO cri-o config, the OCP users can't use OCI artifacts, and we're not going to give the users a way to toggle OCI artifact. The only way to enable it is to change cri-o config directly until it's available.
When we think we're ready for GA (or TP?), we're gonna enable it by changing the crio config template.